cmd/update-main.go

```diff
@@ -68,90 +68,108 @@ const (
 	minioReleaseURL           = "https://dl.minio.io/server/minio/release/" + runtime.GOOS + "-" + runtime.GOARCH + "/"
 )
 
-func getCurrentReleaseTime(minioVersion, minioBinaryPath string) (releaseTime time.Time, err error) {
-	if releaseTime, err = time.Parse(time.RFC3339, minioVersion); err == nil {
+// updateCheckParams - collection of parameters used to check if the
+// version of the current server process is older than the latest
+// available official release. They enable the testability of various
+// routines in the update checking process.
+type updateCheckParams struct {
+	// version of the current server process
+	version string
+	// path to the server binary
+	serverBinaryPath string
+	// file on the server to check for docker env
+	cgroupFile string
+	// URL of release checksum file (contains version of latest
+	// release)
+	releaseChecksumURL string
+	// current process run mode
+	serverMode string
+}
+
+// getCurrentReleaseTime - returns this process's release time.  If it
+// is official minio version, parsed version is returned else minio
+// binary's mod time is returned.
+func (u *updateCheckParams) getCurrentReleaseTime() (releaseTime time.Time,
```

Comment:
In the above code sample, `t2` is not assigned, so it crashes. Seems like an argument to not have pointer receivers in the Go language! Not sure what your point is, because in my code, the struct is initialized in only place in the application (and few more times in tests), in the same file and not used elsewhere. Obviously it cannot be a nil pointer (by trivial inspection).

Doing some research on this, I found review guidelines in the [Go Wiki](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type) - some cherry-picked points that seem relevant (emphasis my own):

> * If the receiver is a large struct or array, a pointer receiver is more efficient. How large is large? Assume it's equivalent to passing all its elements as arguments to the method. If that feels too large, it's also too large for the receiver.
> * If the receiver is a small array or struct that is naturally a value type (for instance, something like the time.Time type), with no mutable fields and no pointers, or is just a simple basic type such as int or string, a value receiver makes sense. A value receiver can reduce the amount of garbage that can be generated; if a value is passed to a value method, an on-stack copy can be used instead of allocating on the heap. (The compiler tries to be smart about avoiding this allocation, but it can't always succeed.) **Don't choose a value receiver type for this reason without profiling first.**
> * Finally, when in doubt, use a pointer receiver.
> 

Though this is not performance sensitive, for curiosity sake, I did some profiling too using this code https://play.golang.org/p/PAuPwIWojW - it does not run in play.golang.org (takes too long), but running on my laptop I get:

```
$ ./testprog 
PointerReceiverResult
2000000000	         0.41 ns/op
ValueReceiverResult
500000000	         3.05 ns/op
```

The program uses a struct that is identical to `updateCheckParams` (has 5 string properties). Clearly the pointer receiver method is faster by like 7X. In summary, it is ok to use pointer receivers here.